cub/test/catch2_test_device_segmented_reduce.cu

```diff
@@ -310,4 +286,110 @@ C2H_TEST("Device fixed size segmented reduce works with all device interfaces",
     // Verify result
     REQUIRE(expected_result == out_result);
   }
+
+// Skip DeviceReduce::Sum tests for extended floating-point types because of unbounded epsilon due
+// to pseudo associativity of the addition operation over floating point numbers
+#if TEST_TYPES != 3
+  SECTION("sum")
+  {
+    using op_t    = ::cuda::std::plus<>;
+    using accum_t = ::cuda::std::__accumulator_t<op_t, input_t, output_t>;
+
+    // Prepare verification data
+    c2h::host_vector<output_t> h_expected_result(num_segments);
+    compute_fixed_size_segmented_problem_reference(
+      in_items, num_segments, segment_size, op_t{}, accum_t{}, h_expected_result.begin());
+
+    // Run test
+    c2h::device_vector<output_t> d_out_result(num_segments);
+    auto d_out_it = unwrap_it(thrust::raw_pointer_cast(d_out_result.data()));
+    device_segmented_sum(d_in_it, d_out_it, num_segments, segment_size);
+
+    c2h::host_vector<output_t> h_out_result(d_out_result);
+    // Verify result
+    REQUIRE(h_expected_result == h_out_result);
+  }
+#endif
+
+  SECTION("min")
+  {
+    using op_t = ::cuda::minimum<>;
+
+    // Prepare verification data
+    c2h::host_vector<output_t> h_expected_result(num_segments);
+    compute_fixed_size_segmented_problem_reference(
+      in_items,
+      num_segments,
+      segment_size,
+      op_t{},
+      ::cuda::std::numeric_limits<input_t>::max(),
+      h_expected_result.begin());
+
+    // Run test
+    c2h::device_vector<output_t> d_out_result(num_segments);
+    auto d_out_it = thrust::raw_pointer_cast(d_out_result.data());
+    device_segmented_min(unwrap_it(d_in_it), unwrap_it(d_out_it), num_segments, segment_size);
+
+    c2h::host_vector<output_t> h_out_result(d_out_result);
+    // Verify result
+    REQUIRE(h_expected_result == h_out_result);
+  }
+
+  SECTION("argmin")
+  {
+    using result_t = ::cuda::std::pair<int, output_t>;
+
+    // Prepare verification data
+    c2h::host_vector<result_t> h_expected_result(num_segments);
+    compute_fixed_size_segmented_argmin_reference(in_items, num_segments, segment_size, h_expected_result.begin());
```

Comment:
Could you make sure that we also test for empty segments (i.e., `segment_size=0`). I think you can just add this to the `GENERATE_COPY` above:

```
const segment_size_t segment_size = GENERATE_COPY(
values({0}),
take(2, random(1 << 0, 1 << 5)),
...
```